extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1252,20 +1255,19 @@ export class AmpA4A extends AMP.BaseElement {
             'height': this.element.getAttribute('height'),
             'width': this.element.getAttribute('width'),
             'src': srcPath,
-            'name': nameData,
           }, SHARED_IFRAME_PROPERTIES));
-      if (method == XORIGIN_MODE.NAMEFRAME) {
         // TODO(bradfrizzell): change name of function and var
-        const attributes = getContextMetadata(
-            this.win, this.element, this.sentinel);
-        attributes['creative'] = creative;
-        const name = JSON.stringify(attributes);
-        // Need to reassign the name once we've generated the context
-        // attributes off of the iframe. Need the iframe to generate.
-        iframe.setAttribute('name', name);
-        iframe.setAttribute('data-amp-3p-sentinel', this.sentinel);
+      let contextMetadata = getContextMetadata(
+          this.win, this.element, this.sentinel);
+      if (method == XORIGIN_MODE.NAMEFRAME) {
+        contextMetadata['creative'] = creative;
```

Comment:
I am a little weary about modifying function results like this. Iframe name construction is already a mess as it is :(

I understand if you need to get this in a canary so LGTM for now, but please add a high priority issue to clean this up. Otherwise, working on this would get very painful in the future.